File: repos_estudiantes_TARQUI_tercera_entrega/ILLANES_ESCOBAR_FERNANDO_JOHSUA/PERSISTENCIA/Ejercicio1/Python/GestorEmpleado.py
from Empleado import Empleado
import pickle
import os

class GestorEmpleado:
    def __init__(self, archivo="empleados.pkl"):
        self.archivo = archivo

    def guardar_empleado(self, emp: Empleado):
        empleados = self.leer_todos()
        empleados.append(emp)
        with open(self.archivo, 'wb') as f:
            pickle.dump(empleados, f)

    def leer_todos(self):
        if not os.path.exists(self.archivo):
            return []
        with open(self.archivo, 'rb') as f:
            return pickle.load(f)

    def busca_empleado(self, nombre: str):
        for emp in self.leer_todos():
            if emp.nombre.lower() == nombre.lower():
                return emp
        return None

    def mayor_salario(self, sueldo: float):
        for emp in self.leer_todos():
            if emp.salario > sueldo:
                return emp
        return None

File: repos_estudiantes_TARQUI_tercera_entrega/ILLANES_ESCOBAR_FERNANDO_JOHSUA/PERSISTENCIA/Ejercicio1/Java/GestorEmpleado.java
package PERSISTENCIA.Ejercicio1.Java;

import java.io.*;
import java.util.ArrayList;

public class GestorEmpleado {
    private final String archivo = "empleados.dat";

    public void guardarEmpleado(Empleado e) {
        ArrayList<Empleado> lista = leerTodos();
        lista.add(e);
        try (ObjectOutputStream out = new ObjectOutputStream(new FileOutputStream(archivo))) {
            out.writeObject(lista);
        } catch (IOException ex) {
            ex.printStackTrace();
        }
    }

    public ArrayList<Empleado> leerTodos() {
        try (ObjectInputStream in = new ObjectInputStream(new FileInputStream(archivo))) {
            return (ArrayList<Empleado>) in.readObject();
        } catch (IOException | ClassNotFoundException e) {
            return new ArrayList<>();
        }
    }

    public Empleado buscaEmpleado(String nombre) {
        for (Empleado e : leerTodos()) {
            if (e.getNombre().equalsIgnoreCase(nombre)) {
                return e;
            }
        }
        return null;
    }

    public Empleado mayorSalario(float sueldo) {
        for (Empleado e : leerTodos()) {
            if (e.getSalario() > sueldo) {
                return e;
            }
        }
        return null;
    }
}
